File: java/src/Structural/Adapter/LooseLeafTea.java
package Structural.Adapter;


//LooseLeafTea.java - the adaptee

public class LooseLeafTea {  
   boolean teaIsSteeped; 
    
   public LooseLeafTea() {
       teaIsSteeped = false;
   }
   
   public void steepTea() {
       teaIsSteeped = true;
       System.out.println("tea is steeping");
   }
}

File: ruby/src/Structural/Adapter/LooseLeafTea.rb

# - the adaptee


class LooseLeafTea

	def initialize()
		@teaIsSteeped = false
	end

	def steepTea()
 		@teaIsSteeped = true
		return 'tea is steeping'
	end

	def getTeaIsSteeped
		@teaIsSteeped
	end

end

